src/monstergenerator.cpp

```diff
@@ -53,6 +54,69 @@ void MonsterGenerator::finalize_mtypes()
     }
 }
 
+void MonsterGenerator::apply_base_faction( const monfaction *base, monfaction *faction )
+{
+    for( const auto &pair : base->attitude_map ) {
+        // Fill in values set in base faction, but not in derived one
+        if( faction->attitude_map.count( pair.first ) == 0 ) {
+            faction->attitude_map.insert( pair );
+        }
+    }
+}
+
+void MonsterGenerator::finalize_monfactions()
+{
+    monfaction *nullfaction = get_or_add_faction( "" );
+    nullfaction->base_faction = nullptr;
+    // Create a tree of faction dependence
+    std::multimap< const monfaction*, monfaction* > child_map;
+    std::set< monfaction* > unloaded; // To check if cycles exist
+    std::queue< monfaction* > queue;
+    for( auto &pair : faction_map ) {
+        auto faction = &pair.second;
+        unloaded.insert( faction );
+        // Point parent to children
+        child_map.emplace( faction->base_faction, faction );
```

Comment:
That's strange

std::multimap::emplace is part of the core std lib and it worked just fine with gcc 4.9.1.

Switched for manual pair creation and then inserting that. It should do the same, it just looks less clean.